File: 2019/question_1.cpp
/* Score: 15/15 */

#include <iostream>
#include <string.h>

using namespace std;

int main() {
    string input = "";
    cin >> input;

    int nums[2][2] = {{1, 2}, {3, 4}};
    int temp[2][2] = {{1, 2}, {3, 4}};

    for(int i = 0; i < input.length(); i++) {
        if (input.at(i) == 'V') { //reflection across y axis
            nums[0][0] = temp[0][1];
            nums[0][1] = temp[0][0];
            nums[1][0] = temp[1][1];
            nums[1][1] = temp[1][0];
        } else { //reflection across x axis
            nums[0][0] = temp[1][0];
            nums[0][1] = temp[1][1];
            nums[1][0] = temp[0][0];
            nums[1][1] = temp[0][1];
        }
        memcpy(temp, nums, sizeof(temp)); //copies nums to temp
    }

    //prints answer
    for(int i = 0; i < 2; i++) {
        for (int j = 0; j < 2; j++) {
            cout << nums[i][j] << " ";
        }
        cout << endl;
    }
    
    return 0;
}

File: 2019/question_1.py
# Score: 15/15

nums = [[1, 2], [3, 4]]
temp_nums = [[1, 2], [3, 4]];

# gets input
transformations = input();

# Applies transofmrations
for i in transformations:
    if i == "V": # reflection accross y axis
        nums[0][0] = temp_nums[0][1]
        nums[0][1] = temp_nums[0][0]
        nums[1][0] = temp_nums[1][1]
        nums[1][1] = temp_nums[1][0]
    else: # reflection accross x axis
        nums[0][0] = temp_nums[1][0]
        nums[0][1] = temp_nums[1][1]
        nums[1][0] = temp_nums[0][0]
        nums[1][1] = temp_nums[0][1]

    # sets temp array equal to main array
    temp_nums[0][0] = nums[0][0]
    temp_nums[0][1] = nums[0][1]
    temp_nums[1][0] = nums[1][0]
    temp_nums[1][1] = nums[1][1]


# displays answer
for i in range(2):
    print(str(nums[i][0]) + " " + str(nums[i][1]))